### PR TITLE
Add compare picture modal to picture page

### PR DESCRIPTION
**Pull request by CFarcy**

### CODE REVIEW

File: src/modules/RA/Picture/components/PictureFields.js [hunk 1]
```diff
@@ -74,18 +134,115 @@ const Br = () => <br />;
...
+                <SideBySideImages pictures={[pic, picturesList[selectedIndex]]} />
+              </>
+            )}
+            <Button
+              variant="outlined"
+              type="button"
+              onClick={() => setShowModal(false)}
+              style={{ marginTop: '10px' }}
+            >
+              {translate('resources.picture.input.close-compare')}
```
Comment (jrmi) [inline src/modules/RA/Picture/components/PictureFields.js]: Use generic close translation instead.

Comment (CFarcy) [reply src/modules/RA/Picture/components/PictureFields.js]: Done.

File: src/modules/RA/Picture/components/PictureGallery.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  },
+
+});
+
+
+const PictureGallery = ({ pictures, selected, setSelected, toLocaleDateString }) => {
+  const classes = useStyles();
+  return (
+    <div className={classes.gallery}>
+      {pictures.length > 0 && pictures.map((picture, index) => (
```
Comment (jrmi) [inline src/modules/RA/Picture/components/PictureGallery.js:30]:
```suggestion
      {pictures.map((picture, index) => (
```

File: src/modules/RA/Picture/components/PictureGallery.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        (index === selected)
+          ? `${classes.galleryItem} image-selected`
+          : classes.galleryItem
+      }
+          onClick={() => setSelected(index)}
+          onKeyPress={() => setSelected(index)}
+        >
+          <img
+            src={picture.file.thumbnail}
+            alt={`Number ${index} from viewpoint`}
```
Comment (jrmi) [inline src/modules/RA/Picture/components/PictureGallery.js:44]: Should be translated ?

File: src/modules/RA/Picture/components/PictureInput.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  const [showModal, setShowModal] = useState(false);
+  const [picturesList, setPicturesList] = useState([]);
+  const [selectedIndex, setSelectedIndex] = useState(null);
+
+  const classes = useStyles();
+  const dataProvider = useDataProvider();
+  const translate = useTranslate();
+  const locale = useLocale();
+
+  const toLocaleDateString = useCallback(dateString => {
```
Comment (jrmi) [inline src/modules/RA/Picture/components/PictureInput.js:107]: You can put that function in utils/date.js

**Review by jrmi — APPROVED** [review]

File: src/modules/RA/Picture/components/PictureGallery.js (hunk 3 above)
Comment (CFarcy) [reply src/modules/RA/Picture/components/PictureGallery.js]: Done.

File: src/modules/RA/Picture/components/PictureGallery.js (hunk 2 above)
Comment (CFarcy) [reply src/modules/RA/Picture/components/PictureGallery.js]: Done.

File: src/modules/RA/Picture/components/PictureInput.js (hunk 4 above)
Comment (CFarcy) [reply src/modules/RA/Picture/components/PictureInput.js]: Done.

**Pull request merged**